Mathlib/Data/Rat/Cast/Defs.lean

```diff
@@ -8,6 +8,7 @@ import Mathlib.Algebra.Group.Commute.Basic
 import Mathlib.Algebra.GroupWithZero.Units.Lemmas
 import Mathlib.Algebra.Order.Field.Rat
 import Mathlib.Data.Int.Cast.Lemmas
+import Mathlib.Data.NNRat.Lemmas
```

Comment:
Given the current design, I don't really want `NNRat` to go first before `Rat`.

Reply:
Sorry, I don't understand this complaint. `NNRat` should always be earlier than `Rat` since it's used in `Semifield` while `Rat` is used in `Field` only.

Oh I guess you mean "the current design of `NNRat` as a subtype of `Rat`" in which case I shall reply "That design needs to be changed for independent reasons anyway, and I am willing to make it happen soon".

Reply:
Subtype of `Rat` is fine in and of itself. But `NNRat.Defs` imports way too much bundled ordered algebra classes to do what it needs to do. I'd prefer that gets fixed before we add more on top of it.

Reply:
`Data.NNRat.Defs` is already imported in `Data.Rat.Cast.Defs` on master, so I do not understand your comment

Reply:
Sorry, I kinda lost this. I already have a PR disconnecting the basics of `NNRat.Defs` #15072 from bundled ordered algebra that this might conflict with. Can we wait until that gets sorted out?